src/screenbot.js

```diff
@@ -50,17 +65,19 @@ var Screenbot = (function Screenbot() {
           console.log("Source: " + _private.source);
 
           _request(_endpoint(command), function(success, response) {
-
-            if (!success) return cb(false, response.error);
+            if (!success || !response.ok) return _handleError({ code: 1, message: response.error }, cb);
 
             if(_private.source && _private.source.readyState !== 2) {
               _private.source.close();
             }
+
             _private.source = new EventSource(_response_endpoint(response.token));
             _private.source.onmessage = function(event) {
               _private.source.close();
-              if(event.data === "0") cb(false);
-              if(event.data !== "0") cb(true, event.data);
+              if(event.data === "0" || !event.data.length) return _handleError({ code: 2, message: 'No data received' }, cb);
```

Comment:
Instead of putting the number 2 here, (and 1 above on line 68) I would have an object enumerating the possible error codes at the very top of the file. Something like:
```
var ERROR_CODES = {
  CLIENT_UNAVAILABLE: 1,
  NO_DATA_RECEIVED: 2
}
```
Then in this line, you could use `code: ERROR_CODES.NO_DATA_RECEIVED`.

Anywhere there are arbitrary numbers or values, it's always good to have them in constants or config files in order to make it obvious how they are related between functions. With the constant, you'll never have to search through your code looking for `2`.